python/GafferUI/Viewer.py

```diff
@@ -240,6 +240,11 @@ def __keyPress( self, widget, event ) :
 						t["active"].setValue( False )
 				return True
 
+			if event.key == "O" and t["active"].getValue() and "orientation" in t :
+				o = t["orientation"].getValue()
+				numPresets = len( [ v for v in Gaffer.Metadata.registeredValues( t["orientation"] ) if v.startswith( "preset:" ) ] )
+				t["orientation"].setValue( ( o + 1 ) % numPresets )
```

Comment:
This doesn't belong in the Viewer, which knows about tools but not about TransformTools. It's a pain to implement any other way though - you'd have to make a custom PlugValueWidget for the orientation plug, and then connect to the signals on the ancestor View, a bit like [ImageViewUI does](https://github.com/GafferHQ/gaffer/blob/main/python/GafferImageUI/ImageViewUI.py#L1022-L1025). Maybe we could add additional `viewer:cyclePresetShortcut` metadata that could be registered on the tool and supported here? Since we don't have any other use cases yet, perhaps the most prudent thing is to add a todo saying that this isn't the right approach for the long term and making that suggestion?

Reply:
I was uneasy about adding something that specific to `Viewer`, but adding the todo sounds like a good middle ground for now. Fixed in https://github.com/GafferHQ/gaffer/pull/5207/commits/c6b8f793d94feea1526781c49fce9a56c172971c